rewrite-java/src/main/java/org/openrewrite/java/UseStaticImport.java

```diff
@@ -61,16 +67,34 @@ public Validated<Object> validate() {
     @Override
     public TreeVisitor<?, ExecutionContext> getVisitor() {
         TreeVisitor<?, ExecutionContext> preconditions = new UsesMethod<>(methodPattern);
-        if (!methodPattern.contains(" *(")) {
-            int indexSpace = methodPattern.indexOf(' ');
+
+        Set<String> methodNameMatchers = new HashSet<>();
+
+        int indexSpace = methodPattern.indexOf(' ');
+        String typeNamePattern = methodPattern.substring(0, indexSpace);
+
+        if (methodPattern.contains(" *(")) {
+            try {
+                Class<?> cls = Class.forName(typeNamePattern);
+                Set<String> methodNames = Arrays.stream(cls.getMethods())
+                        .map(Method::getName)
+                        .collect(Collectors.toSet());
+                methodNameMatchers.addAll(methodNames);
```

Comment:
Hmm; this class-for-name is not likely to work in all environments where we run recipes. There's the Maven and Gradle plugins, Moderne CLI & Platform, third party CLIs like Quarkus & Micronaut, all of which might have a different classpath when they run a recipe as opposed to what's available when you compile a project.

Is there any way we can fix this issue without this type of use?